### PR TITLE
Tests for settings page, downgrade cypress version

### PR DESCRIPTION
**Pull request by jpita**

### CODE REVIEW

File: cypress/page-objects/settings-page.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export class SettingsPage {
+    navigateToSettingsPage(){
```
Comment (stephanebisson) [inline cypress/page-objects/settings-page.js]: Is this being used or do you always use `cy.clickSettingsButton()` directly?

Comment (jpita) [reply cypress/page-objects/settings-page.js]: fixed - moved the method `navigateToSettingsPage()` to searchPage class and use that in the tests

File: cypress/integration/settings-spec.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+describe('settings page', () =>{
+  var firstElementOfTheSettingsMenuList
+  beforeEach(() =>{
+    cy.navigateToHomePage()
+    cy.clickSettingsButton()
+    firstElementOfTheSettingsMenuList = settingsPage.settingsList().children().first()
+  })
+
+  it('should show all buttons', () => {
```
[the remark this answers is not in the document]
Comment (jpita) [reply cypress/integration/settings-spec.js]: fixed

File: cypress/integration/search-spec.js [hunk 3]
```diff
@@ -36,16 +36,4 @@ describe('Article search', () =>{
...
-    searchPage.search("cattle")
-    searchPage.results().first()
-    cy.enter()
-    cy.downArrow()
-    cy.enter()
-    articlePage.title().should('have.text', "Cattle")
-    cy.clickCloseButton()
-    searchPage.results().first()
-      .children().first().next().should('have.class', 'img')
-  })
```
[the remark this answers is not in the document]
Comment (jpita) [reply cypress/integration/search-spec.js]: probably some issues with merges, fixed

File: cypress/integration/settings-spec.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jpita) [reply cypress/integration/settings-spec.js]: fixed

File: cypress/page-objects/settings-page.js (hunk 1 above)
Comment (jpita) [reply cypress/page-objects/settings-page.js]: moved the method `navigateToSettingsPage()`to `searchPage.js` class and use that in the tests

**Comment by hueitan** [comment]
There is also one thing that you can test, but not sure whether this is testable through cypress

for the item `Privacy` and `Term`, when user enter, it will open a new browser link.

**Comment by jpita** [comment]
> There is also one thing that you can test, but not sure whether this is testable through cypress
> 
> for the item `Privacy` and `Term`, when user enter, it will open a new browser link.

https://docs.cypress.io/guides/references/trade-offs.html#Multiple-tabs

I can only check that the link is there but can't check that it opens.
also I was checking the HTML and it doesn't have a link

```<div class="info"><div class="title">Privacy policy</div></div><div class="link"></div>```

**Comment by hueitan** [comment]
Yes, if you see the html element, there is no `<a/>` exist, we need to programmatic trigger to open a new tab. `</a>` doesn't work in non-touch UI device.

If you use the keyboard, enter, then you will get the link open.

**Comment by jpita** [comment]
yes, that's a drawback from Cypress VS Webdriver.
It can't handle more than one tab.

**Comment by jpita** [comment]
can I merge it or it needs another 👍 from @stephanebisson ?

**Comment by stephanebisson** [comment]
You can merge it.

**Pull request merged**